fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -15,44 +15,41 @@
 
 package com.starrocks.connector.jdbc;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.Lists;
-import com.starrocks.analysis.DateLiteral;
-import com.starrocks.analysis.IntLiteral;
+import com.google.common.collect.ImmutableMap;
 import com.starrocks.catalog.Column;
 import com.starrocks.catalog.Database;
 import com.starrocks.catalog.JDBCResource;
 import com.starrocks.catalog.Table;
-import com.starrocks.catalog.Type;
-import com.starrocks.common.DdlException;
 import com.starrocks.connector.ConnectorMetadata;
-import com.starrocks.connector.ConnectorTableId;
 import com.starrocks.connector.PartitionInfo;
-import com.starrocks.connector.PartitionUtil;
 import com.starrocks.connector.exception.StarRocksConnectorException;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.checkerframework.checker.nullness.qual.NonNull;
 
 import java.sql.Connection;
 import java.sql.DriverManager;
-import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.util.List;
 import java.util.Map;
-import java.util.Set;
-import java.util.stream.Collectors;
 
 public class JDBCMetadata implements ConnectorMetadata {
 
     private static Logger LOG = LogManager.getLogger(JDBCMetadata.class);
 
     private Map<String, String> properties;
-    private String catalogName;
-    private JDBCSchemaResolver schemaResolver;
+    private JDBCMetaResolver metaResolver;
+
+    private final @NonNull JDBCAsyncCache<ImmutableMap<String, String>, List<String>> catalogCache
+            = new JDBCAsyncCache<>();
```

Comment:
For final, How about that there are two jdbc catalog?

Reply:
The structure is as follows: each data source has only one jdbc catalog implementation class, and each implementation class contains some cache classes. My understanding is that there is only one jdbc catalog, but have I not understood your question yet?